pkg/util/deployment/deployment.go

```diff
@@ -270,6 +279,16 @@ func waitForReplicaSetUpdated(c clientset.Interface, desiredGeneration int64, na
 	})
 }
 
+func waitForPodsHashPopulated(c clientset.Interface, namespace, name string) error {
+	return wait.Poll(1*time.Second, 1*time.Minute, func() (bool, error) {
+		rs, err := c.Extensions().ReplicaSets(namespace).Get(name)
+		if err != nil {
+			return false, err
+		}
+		return rs.Status.FullyLabeledReplicas == rs.Spec.Replicas, nil
```

Comment:
See ControllerHasDesiredReplicas and ReplicaSetHasDesiredReplicas. We also need observedGeneration to match the generation value we got back after updating the pod template.

Now that observedGeneration is in DeploymentStatus, DeploymentHasDesiredReplicas should do that check, also.

cc @janetkuo